crates/storage/db/src/abstraction/database.rs

```diff
@@ -43,7 +43,7 @@ pub trait Database: for<'a> DatabaseGAT<'a> {
     /// the end of the execution.
     fn update<T, F>(&self, f: F) -> Result<T, Error>
     where
-        F: Fn(&<Self as DatabaseGAT<'_>>::TXMut) -> T,
+        F: FnOnce(&<Self as DatabaseGAT<'_>>::TXMut) -> T,
```

Comment:
Why this change?

Reply:
It is more straightforward as closure is called once. For use case, it allows me to (iirc) to use reference or mut ref inside closure